roofit/xroofit/src/xRooBrowser.cxx

```diff
@@ -19,13 +19,13 @@
 #include "TEnv.h"
 #include "TFile.h"
 #include "RooWorkspace.h"
-#define protected public
 #include "TRootBrowser.h"
-#undef protected
 #include "TGMenu.h"
 #include "TGFileDialog.h"
 #include "TObjString.h"
 
+#define GETPOPUPMENU(b,m) ((TGPopupMenu*)(*(void**)(((unsigned char*)b) + b->Class()->GetDataMemberOffset(#m))))
+
```

Comment:
```C++
namespace {

// Extended interface for TRootBrowser to access protected members
class TRootBrowserInterface : public TRootBrowser {
public:
TGPopupMenu * GetMenuFile() { return fMenuFile; }

};

} // namespace
```

There is an easier way to access `protected` members, without preprocessor macros and ROOT reflection. You can define an "extended" interface class that exposes the protected members, and then `static_cast` to that one.